src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityContext.cs

```diff
@@ -84,15 +84,10 @@ internal sealed class ImmutabilityContext {
 				return true;
 			}
 
-			// Check for [Immutable] etc.
-			if ( GetImmutabilityFromAttributes( type ).HasFlag( kind ) ) {
-				return true;
-			}
-
-			// Check if we were otherwise told that this type is (sufficiently)
-			// immutable.
-			if ( m_extraImmutableTypes.TryGetValue( type, out var info ) && info.Kind.HasFlag( kind ) ) {
-				return true;
+			if( TryGetImmutableTypeInfo( type, out ImmutableTypeInfo info ) ) {
+				if( info.Kind.HasFlag( kind ) ) {
+					return true;
+				}
 			}

```

Comment:
@mthjones this is to support immutable container types better -- we want to ask "how" a type is immutable and then check if that's sufficiently immutable for us. It shouldn't matter if its done with attributes or done via m_extraImmutableTypes; ImmutableTypeInfo abstracts this away.